src/components/Search.vue

```diff
@@ -12,6 +44,22 @@ export default {
       type: String,
       default: null,
     },
+    searchModeOptions: {
+      type: Array,
+      default: () => SearchModeOptions,
+    },
+    searchMode: {
+      type: String,
+      default: '',
```

Comment:
In other places throughout the code, I've used `default: null` to indicate a nullable field, should we do that here as well?

Reply:
I'm going to amend this to say we should do that, after thinking about it more. `type: String` indicates that users of this component must *pass* a string if they intend to use this prop, but `default: null` provides a good out-of-band signal to indicate that no prop was passed at all. I think I like that convention.